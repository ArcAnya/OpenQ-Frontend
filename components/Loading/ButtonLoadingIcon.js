import React from 'react';

const LoadingIcon = (props) => {
	const color = props.bg;
	return (
		<div>
			{props.graph?
				<div className='graph-anim-wrapper'>
					<div className="size-9"><div className="orbit size-8">
						<div className="planet-8"></div>
						<div className="orbit size-7">
							<div className="planet-7"></div>
							<div className="orbit size-6">
								<div className="planet-wrapper">
									<div className="planet-6"></div>
									<div className="planet-4"></div>
								</div>
								<div className="orbit size-5">
									<div className="planet-5"></div>
									<div className="planet-2"></div>
									<div className="orbit size-4">
										<div className="planet-3"></div>
										<div className="orbit size-2">
											<div className="orbit size-1">
											</div>
										</div>
									</div>
								</div>
							</div>
						</div>
					</div>
					</div>
				</div>:
			
				color == 'white' ? (
					<div>
						<svg
							className="h-5 w-5 mr-3 animate-spin
		rounded-full
		h-6
		w-6
		border-t-2 border-b-2 border-purple-500"
							viewBox="0 0 24 24"
						></svg>
					</div>
				) : (
					<div>
						<svg
							className="h-5 w-5 mr-3 animate-spin
				rounded-full
				h-6
				w-6
				border-t-2 border-b-2 border-white"
							viewBox="0 0 24 24"
						></svg>
					</div>
				)}
		</div>
	);
};

export default LoadingIcon;
