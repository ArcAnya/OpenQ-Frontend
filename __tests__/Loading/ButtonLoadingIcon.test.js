import React from 'react';
import ButtonLoadingIcon from '../../components/Loading/ButtonLoadingIcon';
import renderer from 'react-test-renderer';

describe('ButtonLoadingIcon', () => {
  it('should show graph and match DOM Snapshot', () => {
    const tree = renderer.create(<ButtonLoadingIcon graph={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should show purple border when no graph and bg white and match DOM Snapshot', () => {
    const tree = renderer.create(<ButtonLoadingIcon graph={false} bg={'white'} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('should show white border when no graph and bg not white and match DOM Snapshot', () => {
    const tree = renderer.create(<ButtonLoadingIcon graph={false} bg={'other'} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
