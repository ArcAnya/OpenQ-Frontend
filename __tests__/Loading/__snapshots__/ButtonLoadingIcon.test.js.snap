// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLoadingIcon should show graph and match DOM Snapshot 1`] = `
<div>
  <div
    className="graph-anim-wrapper"
  >
    <div
      className="size-9"
    >
      <div
        className="orbit size-8"
      >
        <div
          className="planet-8"
        />
        <div
          className="orbit size-7"
        >
          <div
            className="planet-7"
          />
          <div
            className="orbit size-6"
          >
            <div
              className="planet-wrapper"
            >
              <div
                className="planet-6"
              />
              <div
                className="planet-4"
              />
            </div>
            <div
              className="orbit size-5"
            >
              <div
                className="planet-5"
              />
              <div
                className="planet-2"
              />
              <div
                className="orbit size-4"
              >
                <div
                  className="planet-3"
                />
                <div
                  className="orbit size-2"
                >
                  <div
                    className="orbit size-1"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ButtonLoadingIcon should show purple border when no graph and bg white and match DOM Snapshot 1`] = `
<div>
  <div>
    <svg
      className=" animate-spin rounded-full h-6 w-6 border-t-2 border-b-2 border-purple-500"
      data-testid="loading"
      viewBox="0 0 24 24"
    />
  </div>
</div>
`;

exports[`ButtonLoadingIcon should show white border when no graph and bg not white and match DOM Snapshot 1`] = `
<div>
  <div>
    <svg
      className=" animate-spin rounded-full h-6 w-6 border-t-2 border-b-2 border-white"
      data-testid="loading"
      viewBox="0 0 24 24"
    />
  </div>
</div>
`;
